lib/config-generator.js

```diff
@@ -100,6 +102,24 @@ class ConfigGenerator {
             alias: {}
         };
 
+        if (this.webpackConfig.useVueLoader && this.webpackConfig.vueOptions.runtimeCompilerBuild) {
+            if (this.webpackConfig.isProduction()) {
+                logger.recommendation(`If you do not need to compile Vue templates at runtime, pass ${chalk.green('{ runtimeCompilerBuild: false }')} as the 3rd argument to ${chalk.green('enableVueLoader()')} for a smaller build.`);
+            }
+
+            const vueVersion = getVueVersion(this.webpackConfig);
+            switch (vueVersion) {
+                case 2:
+                    config.resolve.alias['vue$'] = 'vue/dist/vue.esm.js';
+                    break;
+                case 3:
+                    config.resolve.alias['vue'] = 'vue/dist/vue.esm-bundler.js';
```

Comment:
I think we also need to alias only `vue$` here, to avoid breaking cases where they import `vue/dist/*` explicitly.